hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs

```diff
@@ -106,6 +107,20 @@ runActionWithContext df (Just uri) def action = do
   mcradle <- getCradle uri
   loadCradle df mcradle def action
 
+-- ---------------------------------------------------------------------
+
+runWithContext :: Monoid a => Uri -> IdeGhcM (IdeResult a) -> IdeGhcM (IdeResult a)
+runWithContext uri act = case uriToFilePath uri of
```

Comment:
I'm not really a fan of this function because it calls `error`. Is there an issue using `runActionWithContext`? 

Why you would ever want to use this function should be documented anyway as I don't think it should ever be used in the `h-i-e` code base.

Reply:
This function was moved here from the `test/unit/HaRePluginSpec.hs`, so that it can be available for use in other modules, and is used in HaRe for running refactors.

I will see what other options we have.

Reply:
Clearly documenting it's purpose and intended usage is fine in my opinion. It mainly should be clear, that neither hie, nor hie-plugin-api is allowed to really use this function.